ui/prometheus-plugin/src/utils/utils.ts

```diff
@@ -74,3 +74,53 @@ export function formatSeriesName(inputFormat: string, seriesLabels: SeriesLabels
   const resolveLabelsRegex = /\{\{\s*(.+?)\s*\}\}/g;
   return inputFormat.replace(resolveLabelsRegex, (_, g) => (seriesLabels[g] ? seriesLabels[g] : g));
 }
+
+/*
+ * Stringifies object of labels into valid PromQL for querying metric by label
+ */
+function stringifyPrometheusMetricLabels(labels: { [key: string]: unknown }, removeExprWrap?: boolean) {
+  const labelStrings: string[] = [];
+  Object.keys(labels)
+    .sort()
+    .forEach((labelName) => {
+      const labelValue = labels[labelName];
+      if (typeof labelValue === 'string') {
```

Comment:
Does this mean we'd skip values that are numbers?

Reply:
So that shouldn't ever happen with what Prom supports in metric label values, but changing this to check whether labelValue is undefined seems to work fine too so just updated this!